### PR TITLE
AWS Batch resources are added.

### PR DESCRIPTION
**Pull request by tahir-duplo**
- Resource `duplocloud_aws_batch_scheduling_policy` added.
- Resource `duplocloud_aws_batch_compute_environment` added.
- Resource `duplocloud_aws_batch_job_queue` added.
- Resource `duplocloud_aws_batch_job_definition` added.

### CODE REVIEW

**Comment by zafarabbas** [comment]
Task linked: [DUPLO-7525 TF: Support for AWS Batch](https://app.clickup.com/t/8655600/DUPLO-7525)

File: docs/resources/aws_batch_compute_environment.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+- **status** (String) The current status of the compute environment (for example, CREATING or VALID).
+- **status_reason** (String) A short, human-readable string to provide additional details about the current status of the compute environment.
+
+<a id="nestedblock--compute_resources"></a>
+### Nested Schema for `compute_resources`
+
+Required:
+
+- **max_vcpus** (Number) The maximum number of EC2 vCPUs that an environment can reach.
+- **type** (String) The type of compute environment. Valid items are `EC2`, `SPOT`, `FARGATE` or `FARGATE_SPOT`.
```
Comment (zafarabbas) [inline docs/resources/aws_batch_compute_environment.md:83]: do all these types work right now?

File: docs/resources/aws_batch_compute_environment.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- **name** (String) Specifies the name of the compute environment.
+- **tenant_id** (String) The GUID of the tenant that the aws batch compute environment will be created in.
+- **type** (String) The type of the compute environment. Valid items are `MANAGED` or `UNMANAGED`.
```
[the remark this answers is not in the document]
Comment (tahir-duplo) [reply docs/resources/aws_batch_compute_environment.md:57]: @zafarabbas I have exposed all supported parameters from AWS. We should keep these and pass only `MANAGED` type.

File: docs/resources/aws_batch_compute_environment.md (hunk 1 above)
Comment (tahir-duplo) [reply docs/resources/aws_batch_compute_environment.md:83]: Yes, All above are supported from Duplo.

**Review by zafarabbas — APPROVED** [review]

**Pull request merged**